x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,248 @@
+  logger: Logger,
+  params: ConnectorParamsType,
+  variables: Record<string, unknown>
+): ConnectorParamsType {
+  return {
+    connector: params.connector,
+    message: params.message,
+    rule: params.rule,
+    alertState: params.alertState,
+  };
+}
+
+async function executor(
+  execOptions: ObsAIAssistantConnectorTypeExecutorOptions,
+  initResources: (request: KibanaRequest) => Promise<ObservabilityAIAssistantRouteHandlerResources>
+): Promise<ConnectorTypeExecutorResult<unknown>> {
+  const state = execOptions.params.alertState;
+  if (state !== 'new') {
+    // system connector action frequency can't be user configured. we use this
+    // flag as dedup mechanism to prevent triggering the same worfklow again.
+    // while recovered state is a valid trigger we only allow new alerts for now
+    return { actionId: execOptions.actionId, status: 'ok' };
+  }
+
+  const request = execOptions.request!;
+  const resources = await initResources(request);
+  const client = await resources.service.getClient({ request });
+  const functionClient = await resources.service.getFunctionClient({
+    signal: new AbortController().signal,
+    resources,
+    client,
+    screenContexts: [],
+  });
+  const actionsClient = await (
+    await resources.plugins.actions.start()
+  ).getActionsClientWithRequest(request);
+
+  const systemMessage = buildSystemMessage(functionClient, execOptions.params.rule, state);
+
+  await lastValueFrom(
+    client
+      .complete({
+        functionClient,
+        persist: true,
+        isPublic: true,
+        connectorId: execOptions.params.connector,
+        signal: new AbortController().signal,
+        kibanaPublicUrl: (await resources.context.core).coreStart.http.basePath.publicBaseUrl,
+        messages: [
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.System,
+              content: systemMessage,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              content: execOptions.params.message,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.Assistant,
+              content: '',
+              function_call: {
+                name: 'get_connectors',
+                arguments: JSON.stringify({}),
+                trigger: MessageRole.Assistant as const,
+              },
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              name: 'get_connectors',
+              content: JSON.stringify({
+                connectors: await actionsClient.getAll().then((connectors) => {
+                  return connectors.map((connector) => {
+                    if (connector.actionTypeId === '.slack') {
```

Comment:
I believe  the new `.slack-api` connector is being used a lot by customers, which is a little more useful than the `.slack` one - in particular, it allows posting to more than just one channel (unlike the `.slack` connector), but also requires the channel id's (not names!) to be passed as a param.  Also supports BlockKit, for enhanced formatting.

Reply:
See https://github.com/elastic/kibana/pull/179980#discussion_r1562274152 for context on using the slack webhook. We'll look into implementing other connectors as a follow up